### ISSUE TITLE
Disable .ps1 support in entryPoint

### ISSUE THREAD

**Issue by daxian-dbw** [Azure/azure-functions-powershell-worker#73]
When working on the refactoring, I found that although `Import-Module -Name script.ps1` does import the script file as a module, the module doesn't expose anything, and functions from the script are actually loaded in the global scope.

This unexpected behavior will leave those functions around to pollute the Runspace.
~We could use `New-Module -ScriptBlock` to replace the current approach.~
`New-Module` creates a dynamc module that will be persisted in the session until the session exits. So it doesn't help.

**Comment by daxian-dbw** [comment]
I guess our choices are:
- Don't support `entrypoint` at all.
- Copy the `.ps1` file to a temp file and rename the extension to `.psm1`, then `import-module` the `.psm1` file. This doesn't seem worth the effort.

I'm inclined to the first option. @tylerl0706 what's your thought?

**Comment by tylerl0706** [comment]
Entrypoint is a functions concept and I think we should honor it if we can.

cc @asavaritayal @joeyaiello

**Comment by tylerl0706** [comment]
If we dot source and then run the function in the local scope, wouldn't that work?

```
pwsh.AddScript(". ./foo.ps1; Invoke-Foo", useLocalScope: true).Invoke()
```

**Comment by daxian-dbw** [comment]
Then we cannot use `AddCommand`, which means we cannot differentiate between a hard exception (the script is broken) and an expected error because all kinds of errors will be written to error stream.

**Comment by daxian-dbw** [comment]
> Entrypoint is a Functions concept and I think we should honor it if we can.

I think it depends on whether it adds any value. What would be the scenarios for people to use an entry point along with a `.ps1` file? A `.ps1` file always has a default entry point.

**Comment by tylerl0706** [comment]
I thought we had `pwsh.HasErrors` available to us? Or is that set to true when there are errors in the error stream?

Also, using `AddCommand(...).AddStatement()` also has this problem with hard exceptions.

**Comment by daxian-dbw** [comment]
As far as I understand, `pwsh.HasErrors` will be set to `true` if there is any error written to the error stream, which means the expected error will make it set to `true` too.

Didn't know the same issue happens to `AddStatement()` too. That's a shame 😦

**Comment by daxian-dbw** [comment]
> `pwsh.AddScript(". ./foo.ps1; Invoke-Foo", useLocalScope: true).Invoke()`

By the way, doing this will make passing the arguments hacky and error prone.

**Comment by tylerl0706** [comment]
So get this, that actually does throw it looks like? 
![screenshot_20181008-171704](https://user-images.githubusercontent.com/2644648/46639958-19605080-cb1e-11e8-94a3-f5f642044855.jpg)

**Comment by tylerl0706** [comment]
Same with dot sourcing actually.

![screenshot_20181008-180451](https://user-images.githubusercontent.com/2644648/46641120-b4f4bf80-cb24-11e8-968a-b919cf842444.jpg)

**Comment by tylerl0706** [comment]
I might be holding on to nothing here.

Feel free to change it (remove the EntryPoint support) if you think this is something user's won't want and we can get feedback later.

Please confirm with PMs though.

cc @asavaritayal @joeyaiello

**Comment by tylerl0706** [comment]
ping PMs @asavaritayal @joeyaiello

**Comment by tylerl0706** [comment]
I just saw that we got some chatter about classes support. If we keep entryPoint, we can leverage it with classes no? That seems like a reasonable use case?

**Comment by joeyaiello** [comment]
I'm not sure we care about entrypoint right now, but I'll spin up an issue to discuss it, and then we can close this out if we determine it's useless over there.

**Comment by daxian-dbw** [comment]
Closed via #116
